buildingmotif-app/src/app/model-validate/model-validate.component.ts

```diff
@@ -63,13 +68,13 @@ export class ModelValidateComponent implements OnInit{
     if (this.libraries == undefined) return;
 
     const selectedLibraries = this.libraries.filter((_, i) => this.selectedLibrariesForm.value[i])
-    const args = selectedLibraries.map(l => {return l.id})
+    const args = selectedLibraries.map(l => l.id);
 
     if (!!this.modelId){
       this.showValidatingSpinner = true;
 
       this.modelValidateService.validateModel(this.modelId, args).subscribe(
-        res => {this.validationResponse = res},
+        res => {this.validationResponse = JSON.parse(res)},
```

Comment:
If you type ``, you won't have to parse and it'll be safer

```
validateModel(modelId: number, args: number[]) {
   ...
    return this.http.post<ValidationResponse>(...)
   ...
  }
```

Reply:
I'm not sure what this means -- can you clarify?

Reply:
lol, sorry, I forgot to fill the ``. if you type the return statement of `modelValidateService.validateModel` as above, you won't have to parse to json

Reply:
Great to know this is possible! Thanks